web-app/src/main/webapp/script.js

```diff
@@ -59,8 +59,21 @@ function showMessageOnInfoWindow(message, position, map, infoWindow) {
 window.onload = () => {
   document.getElementById("form-container").style.display = "none";
   document.getElementById('report-button').addEventListener('click', showReportForm);
+  setLoginStatus();
```

Comment:
I'm far from a JavaScript expert so I may well be wrong about this.
What will happen if you call an async function synchronously like that?

Perhaps you need to do instead:
```js
window.onload = async () => {
  blah();
  await setLoginStatus();
};
```